File: Backend/cloud/app/schemas/resource.py
"""Pydantic schemas for Cloud Resource responses."""
from __future__ import annotations

import uuid
from datetime import datetime
from typing import Any, Dict, Optional

from pydantic import BaseModel


class CloudResourceResponse(BaseModel):
    id: uuid.UUID
    account_id: uuid.UUID
    provider_resource_id: str
    resource_type: str
    resource_name: str
    region_or_zone: str
    status: Optional[str] = None
    ip_address: Optional[str] = None
    config: Optional[Dict[str, Any]] = None
    metadata_: Optional[Dict[str, Any]] = None
    cost_monthly: Optional[float] = None
    tags: Optional[Dict[str, Any]] = None
    discovered_at: datetime

    model_config = {"from_attributes": True}


class CloudResourceDetail(CloudResourceResponse):
    raw_data: Optional[Dict[str, Any]] = None

File: Actmon_V1/src/features/cloud/types/resource.ts
export interface CloudResource {
  id: string;
  account_id: string;
  provider_resource_id: string;
  resource_type: string;
  resource_name: string;
  region_or_zone: string;
  status: string | null;
  ip_address?: string | null;
  cost_monthly?: number | null;
  discovered_at: string;
  tags?: Record<string, string> | null;
  config?: Record<string, any> | null;
}

export interface CloudResourceDetail extends CloudResource {
  metadata_?: Record<string, any> | null;
  raw_data?: Record<string, any> | null;
}
